### PR TITLE
Add days badge

### PR DESCRIPTION
**Pull request by tuminzee**
**Proposal/Idea** 
Add days badge to count consistent days of progress.
Whenever a repo is linked/created observer the date. Display badge with date difference of start date and end date 

> start date - date when repo is linked/created.
> end date - current date atm.

image
image

**Implementation**
1. Add `checkbox` in `welcome.html` which determines the choice of display of icon badge.
2. Changes in the checkbox triggers and updates the value of `showBadge`, `startDate` values in local storage.
3. Add alarm to trigger update badge after every minute.
4. Add event listeners to handle edge cases.

**Tested Edge Cases**
Test Environment: Local 
Branch: dev
Browser version: Version 96.0.4664.110 (Official Build) (64-bit)
System: PopOs 20.04 LTS

1. What happens when the extension is installed?
- `showBadge` and `startDate` are set to its default values `[true, Date.now()]`

2. What happens when a day changes?
- `alarm trigger` updates the value of the badge after every minute; though latency can be expected.
- `icon badge` might not be exactly updated at 00:00

3. What is the state of checkbox?
- It is interpolated with `showBadge` value, so the choice will be remembered.

4. A trigger every minutes, isn't that insane?
- Yes, it is. I plan to set that value to 5 mins, or 30 mins in the future. 
- Alongside I am also planning a `commit trigger` to update the badge/

5. What happens when a repo in unlinked?
- `icon badge` is cleared, `showDate` is set to `false`, `startDate` is removed from the `local storage`

**Untested Edge Cases**
1. What happens when there is an error in the flow of auth?
2. How much is the performance cost in setting the `alarm trigger` to 1 min in `background.js`

**Note**
1. Days difference is absolute and 0-indexed.
2. All the changes are console logged atm.
2. Trigger structure needs improvement.
3. Design needs improvement. 

**Upcoming collaborations**
1. Consistent usage of `ES6`.
2. Consistent usage of `jquery`, or `js dom`.
3. Add days in the commit message. 
4. Add `custom-date` choice for `startDate`. 
5. Add linting
6. Add github actions
7. Add `CODE OF CONTRIBUTION`

### CODE REVIEW

File: scripts/background.js [hunk 1]
```diff
@@ -1,3 +1,79 @@
+function setStartDate(){
+  const currentTime = (new Date()).toJSON();
+  const items = { 'start_date': currentTime }; 
+  chrome.storage.local.set(items, () => {
+      if (chrome.runtime.lastError) {
+          console.error(chrome.runtime.lastError.message);
+      }
+      else{
+        console.log('set start_date:' + currentTime);
```
Comment (andythsu) [inline scripts/background.js:9]: are these `console.log` necessary for prod?

Comment (tuminzee) [reply scripts/background.js:9]: Not at all. I added it because I couldn't find a way to notice all the changes.

File: scripts/background.js [hunk 2]
```diff
@@ -1,3 +1,79 @@
+function clearBadge() {
```
Comment (QasimWani) [inline scripts/background.js:1]: could we split all the notification code into a separate file? let's call it `notification.js`

**Review by QasimWani — CHANGES_REQUESTED** [review]
@tuminzee can u take a look at recent main branch and make another PR with latest changes. there are about 10 merge conflicts in leetcode.js. Most of them are trivial, but if you can make another PR with latest changes, that'd make things quicker.

**Comment by tuminzee** [comment]
@QasimWani please have a look
